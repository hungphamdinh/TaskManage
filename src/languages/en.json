{
  "common": {
    "app_name": "React Native Base",
    "money": "đ",
    "cancel": "Huỷ bỏ",
    "confirm": "Xác nhận",
    "date_time": {
      "month": "Tháng",
      "day": "Ngày",
      "hour": "giờ",
      "minute": "phút"
    }
  },
  "login_screen": {
    "title": "Login Screen",
    "desc": "Login with email and password",
    "username": "U s e r n a m e",
    "password": "P a s s w o r d",
    "button": "Login",
    "forgot_password": "Forgot password?"
  },
  "home_screen": {
    "title": "Home Screen",
    "logout_button": "Logout"
  },
  "warning": {
    "missing_info": "Vui lòng điền đầy đủ thông tin"
  }
}
