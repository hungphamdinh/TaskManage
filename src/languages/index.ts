import strings from './vi';

export { strings };
