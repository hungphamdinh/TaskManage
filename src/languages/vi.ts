export default {
  common: {
    app_name: 'Find Work',
    company_name: 'Adwardsoft Co.,Ltd',
    currency: 'đ',
    currency_vnd: 'VND',
    button_next: 'Tiếp tục',
    cancel: 'Cancel',
    confirm: 'Xác nhận',
    blank_input: ' *',
    date_time: {
      month: 'Tháng',
      day: 'Ngày',
      hour: 'giờ',
      minute: 'phút',
    },
    yes: 'Có',
    no: 'Không',
    delete: 'Delete',
    accept: 'Đồng ý',
    warning: 'Warning',
    vnd: 'đ',
    month: ' /chuyến',
    home: 'Trang chủ',
    contract: 'Hợp đồng',
    search: 'Tìm kiếm',
  },
  auth_info: {
    email: 'Email',
    password: 'Password',
    confirm_password: 'Confirm Password',
    full_name: 'Full Name',
    email_address: 'Email Address',
    phone_number: 'Phone Number',    
    login: 'Login',
    sign_up: 'Sign Up',
    warning: {
      email: 'Please enter valid email',
      email_required: 'Email Address is Required',
      password_least: 'Password must be at least',
      characters: 'characters',
      password_required: 'Password is Required',
    }
  },
  board_screen: {
    task: 'Task',
    today: 'Today',
    add_task: 'Add task',
  },
  warning_api: {
    login_failed: 'Đăng nhập thất bại',
  },
  login_screen: {
    title: 'Manage\nYour task easily',
    sub_title: 'Easy wayt to manage daily task and office management',
  },
};
