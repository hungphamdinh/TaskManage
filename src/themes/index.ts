import Colors from './Colors';
import Fonts from './Fonts';
import Metrics from './Metrics';
import Images from './Images';
import Styles from './Styles';
// import iPhoneXHelper from './iPhoneXHelper';
import { IconSize } from './Images';
export { Colors, Fonts, Images, Metrics, Styles, IconSize };
