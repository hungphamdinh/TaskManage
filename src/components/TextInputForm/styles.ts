import { StyleSheet } from 'react-native';
import { Colors, Metrics, Fonts } from '../../themes';
import { androidOS } from '../../helpers/Constants';

const styles = StyleSheet.create({
  textInput: {
    fontSize: Fonts.size.large,
  },
  textTextInput: {
    color: Colors.appGrayColor,
    marginBottom: Metrics.margin.small,
  },
  textValidate: {
    color: Colors.appColor,
  },
  circle: {
    width: 4,
    height: 4,
    borderRadius: Metrics.borderRadius.large,
    backgroundColor: Colors.appWhite,
    marginTop: Metrics.margin.tiny,
  },
  dropdown: {
    backgroundColor: Colors.appWhite,
    padding: Metrics.margin.regular,
    marginTop: Metrics.margin.regular,
    position: 'absolute',
    right: 30,
    top: 70,
    zIndex: 2,
  },
  buttonSetting: {
    marginRight: Metrics.margin.small,
    width: 10,
  },
});

export default styles;
