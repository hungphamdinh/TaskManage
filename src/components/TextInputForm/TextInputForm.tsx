import React, { useState } from 'react';
import { View } from 'react-native';
import { TextInput, Text } from 'react-native';
import { Colors } from '../../themes';
import styles from './styles';
import AppText from '../AppText';
import { strings } from '../../languages';
import { Metrics } from '../../themes';
export default (props: any) => {
  const { inputRef, label, isValidate = false } = props;
  const [isFocus, setIsFocus] = useState(false);
  const _onFocus = () => {
    setIsFocus(true);
  };

  const _onBlur = () => {
    setIsFocus(false);
  };
  function borderProps() {
    if (isFocus) {
      return {
        marginTop: Metrics.margin.large,
        marginBottom: Metrics.margin.small,
        paddingBottom: Metrics.margin.regular,
        borderBottomWidth: 2,
        borderBottomColor: Colors.overlay5,
      };
    } else {
      return {
        marginTop: Metrics.margin.large,
        marginBottom: Metrics.margin.small,
        paddingBottom: Metrics.margin.regular,
        borderBottomWidth: 1,
        borderBottomColor: Colors.appGrayColor,
      };
    }
  }
  let lineStyle = {
    ...borderProps(),
  };
  return (
    <View style={lineStyle}>
      <Text style={styles.textTextInput}>
        <AppText text={label} color={Colors.appGrayColor} />
        {isValidate ? (
          <AppText
            style={styles.textValidate}
            text={strings.common.blank_input}
          />
        ) : null}
      </Text>

      <TextInput
        onFocus={_onFocus}
        onBlur={_onBlur}
        {...props}
        style={styles.textInput}
        ref={inputRef}
      />
    </View>
  );
};
