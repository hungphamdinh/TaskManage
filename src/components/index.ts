import AppText from "./AppText";
import TextInputForm from "./TextInputForm";
import AppButton from "./AppButton";
import Divider from "./Divider";
import Background from "./Background/Background";
import Indicator from './Indicator';
import AlertDialog from './AlertDialog';
import ModalInput from './ModalInput';
export { AppText, TextInputForm, AppButton, Divider, Background, Indicator, AlertDialog, ModalInput};
