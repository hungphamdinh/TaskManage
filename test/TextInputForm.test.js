import React from "react";
import renderer from "react-test-renderer";
import { cleanup, render, fireEvent } from "@testing-library/react-native";
import TextInputForm from "../src/components/TextInputForm";
import { strings } from "../src/languages";
describe("TextInputForm", () => {
  test("renders correctly", () => {
    const tree = renderer.create(<TextInputForm />).toJSON();
    // console.log(tree.props);
    expect(tree).toMatchSnapshot();
  });

  test("Change Text Input TextInputForm", () => {
    const onChangeTextMock = jest.fn();
    const page = render(
      <TextInputForm
        placeholder={strings.add_member_screen.search}
        onChangeText={onChangeTextMock}
      />
    );
    const inputSearch = page.getByPlaceholderText(
      strings.add_member_screen.search
    );
    fireEvent.changeText(inputSearch, "Hell323o");
    expect(onChangeTextMock).toHaveBeenCalled();
  });
});
afterEach(cleanup);

// it("Change Text Function", () => {
//   const instanceOf = renderer.create(<TextInputForm />);
//   console.log(instanceOf)
//   // instanceOf._onFocus();
//   expect(instanceOf.state.isFocus).toEqual(true);
// });
