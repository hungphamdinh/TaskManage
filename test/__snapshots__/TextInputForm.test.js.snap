// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputForm renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "rgba(0,0,0,0.1)",
      "borderBottomWidth": 1,
      "marginBottom": 5,
      "marginTop": 15,
      "paddingBottom": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#919BB3",
        "marginBottom": 5,
      }
    }
  >
    <Text
      allowFontScaling={true}
      color="#919BB3"
      style={
        Array [
          Object {
            "color": "#919BB3",
            "fontSize": 13,
            "fontStyle": "normal",
            "fontWeight": "400",
            "textAlign": "left",
          },
          undefined,
        ]
      }
    />
  </Text>
  <TextInput
    allowFontScaling={true}
    editable={true}
    onBlur={[Function]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    selectTextOnFocus={true}
    style={
      Object {
        "fontSize": 16,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
